src/libstd/io/stdio.rs

```diff
@@ -86,6 +86,41 @@ impl Write for StderrRaw {
     fn flush(&mut self) -> io::Result<()> { Ok(()) }
 }
 
+enum MaybeWrite<W> {
+    Real(W),
+    Sink,
+}
+
+impl<W: io::Write> io::Write for MaybeWrite<W> {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        match *self {
+            MaybeWrite::Real(ref mut w) => w.write(buf),
+            MaybeWrite::Sink => Ok(buf.len())
+        }
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        match *self {
+            MaybeWrite::Real(ref mut w) => w.flush(),
+            MaybeWrite::Sink => Ok(())
+        }
+    }
+}
+
+enum MaybeRead<R> {
+    Real(R),
+    Empty,
+}
```

Comment:
Perhaps this could just be:

```rust
enum Maybe<T> {
    Real(T),
    Empty,
}

impl<W: Write> Write for Maybe<W> { ... }
impl<R: Read> Read for Maybe<W> { ... }
```